### PR TITLE
fixes #251 table_exists? deprecation warning with Rails 5.0

### LINKED ISSUE #251: DEPRECATION WARNING on rails 5.0 as of acts_as_list 0.9
Since upgrading to `0.9.0`, I've been seeing this deprecation error on Rails 5.0.1:

> DEPRECATION WARNING: #table_exists? currently checks both tables and views. This behavior is deprecated and will be changed with Rails 5.1 to only check tables. Use #data_source_exists? instead.

The call to table_exists was added [here](https://github.com/swanandp/acts_as_list/commit/cb8dfb58824af0663ad5e5b113bd36d69954ce05#diff-b505f02d01ecca97ea8fa490cc4ae70aR7)

### PR DESCRIPTION
**Pull request by zharikovpro**

### CODE REVIEW

File: lib/acts_as_list/active_record/acts/sequential_updates_method_definer.rb [hunk 1]
```diff
@@ -4,7 +4,12 @@ def self.call(caller_class, column, sequential_updates_option)
       define_method :sequential_updates? do
         if !defined?(@sequential_updates)
           if sequential_updates_option.nil?
-            table_exists = caller_class.connection.table_exists?(caller_class.table_name)
+            table_exists =
+              if defined?(caller_class.connection.data_source_exists?)
```
Comment (brendon) [inline lib/acts_as_list/active_record/acts/sequential_updates_method_definer.rb]: I think I'd rather be more explicit about why we're forking in the code here. Can we use `if ActiveRecord::VERSION::MAJOR >= 5` instead? Then we know we can remove this in the future if we drop Rails < 5 support.

**Comment by brendon** [comment]
Thanks @zharikovpro :) Just one tweak and we'll be done with this. I searched and we're doing the same thing in the test harness too (`def teardown`).

**Comment by zharikovpro** [comment]
@brendon sure, done.

**Comment by brendon** [comment]
Thanks @zharikovpro :) Great work as usual!

**Pull request merged**